### ISSUE TITLE
POST to webhook source can fail with 404 without error message

### ISSUE THREAD

**Issue by def-** [MaterializeInc/materialize#23027]
### What version of Materialize are you using?

b2941b38a5

### What is the issue?

Seen when trying to extend webhook sources: https://buildkite.com/materialize/nightlies/builds/5065#_
```
ValueError: POST failed: 404,
```
Normally there would be a `no object was found at the path` error message, but it's missing in this case.

What is special is that this happened in parallel-workload with naughty identifiers and renames, I'm not sure if it's related but I can imagine that this can happen when you rename a schema and then try to POST to a webhook source in it shortly after.

**Comment by ParkMyCar** [comment]
I looked into this and I think figured out the issue.

I was able to repro this locally using:
```
bin/mzcompose --find parallel-workload down && bin/mzcompose --find parallel-workload run default --seed=1699439746 --threads=32 --runtime=1500 --complexity=ddl --scenario=rename --naughty-identifiers
```
which seemed to be the same config as the failing CI run.

In https://github.com/MaterializeInc/materialize/pull/23043 I added some debug logging that prints the failing URL, which I found to be:
```
http://localhost:6976/api/webhook/<materialize.parallel_workload.database.Database object at 0x107200b10>/public/"db-pw-1699439746-0_̡͓̞ͅI̗̘̦"."s-0-2_0.0/0.0"."wh-0_￾"
```
What seems to be the problem is the generated URL contains an extra `/` character in the source name, i.e. the `/0.0"."wh-0_￾"` portion at the very end, which gets parsed as a 4th URL component. So when POSTing to this URL it's not getting handled by our webhook logic, which only matches on 3 components, instead the axum HTTP server is correctly rejecting it with 404 since we don't have a handler for `/api/webhook/1/2/3/4` routes.
